src/main/java/com/icthh/xm/tmf/ms/communication/messaging/MessagingHandler.java

```diff
@@ -34,15 +34,16 @@ public void receiveMessage(CommunicationMessage message) {
         List<String> phoneNumbers = new ArrayList<>(from(message).getPhoneNumbers());
         for (String phoneNumber : phoneNumbers) {
             try {
+                businessRuleValidator.validate(message);
```

Comment:
let me propose not to use exception as business rule result. As Exception throwing is quite expensive in java and this business rules are tend to be invoked in high performance manner. I think it is better to return the result and check it without throwing an exception.

Reply:
In this case rule interrupted send message, and cpu overhead so small, but so more profit by readability